subgraphs/euler-finance/src/mappings/handlers.ts

```
@@ -293,3 +324,110 @@ export function handleGovSetReserveFee(event: GovSetReserveFee): void {
+    if (prevEpoch) {
+      // finalize mkt._weightedStakedAmount and mrkt._weightedTotalBorrowUSD
+      // epoch.top10StakeAmounts for prev Epoch
+      const marketStakeAmounts: BigInt[] = [];
+      let sumWeightedBorrowUSD = BIGDECIMAL_ZERO;
+      for (let i = 0; i < protocol._marketIDs!.length; i++) {
+        const mktID = protocol._marketIDs![i];
+        const mrkt = getOrCreateMarket(mktID);
+        const stakedAmount = mrkt._stakedAmount;
+        if (stakedAmount.gt(BIGINT_ZERO)) {
+          updateWeightedStakeAmount(mrkt, epochStartBlock);
+          mrkt._stakeLastUpdateBlock = epochStartBlock;
+        }
+        marketStakeAmounts.push(mrkt._weightedStakedAmount ? mrkt._weightedStakedAmount! : BIGINT_ZERO);
+        if (mrkt.totalBorrowBalanceUSD.gt(BIGDECIMAL_ZERO)) {
+          updateWeightedBorrow(mrkt, prevEpoch, epochStartBlock);
+        }
+        sumWeightedBorrowUSD = sumWeightedBorrowUSD.plus(
+          mrkt._weightedTotalBorrowUSD ? mrkt._weightedTotalBorrowUSD! : BIGDECIMAL_ZERO,
+        );
+        mrkt.save();
+      }
+
+      const eulerContract = Euler.bind(Address.fromString(EULER_ADDRESS));
+      const execProxyAddress = eulerContract.moduleIdToProxy(MODULEID__EXEC);
+      // update EUL token prices
+      updatePrices(execProxyAddress, market, event);
+
+      const cutoffAmount = getCutoffValue(marketStakeAmounts, 10);
+      const totalRewardAmount = BigDecimal.fromString((EUL_DIST[prevEpochID - START_EPOCH] * EUL_DECIMALS).toString());
+
+      const EULToken = getOrCreateToken(Address.fromString(EUL_ADDRESS));
+      const rewardToken = getOrCreateRewardToken(Address.fromString(EUL_ADDRESS), RewardTokenType.BORROW);
+      const dailyScaler = BigDecimal.fromString((BLOCKS_PER_DAY / (BLOCKS_PER_EPOCH as f64)).toString());
+      for (let i = 0; i < protocol._marketIDs!.length; i++) {
+        const mktID = protocol._marketIDs![i];
+        const mrkt = getOrCreateMarket(mktID);
```

Kinda the same idea with what you mentioned on my PR. My take is that it is supposed to be there, so if it is not you would want to throw an error and skip

---

To clarify, are you referring to `mrkt = getOrCreateMarket(mktID)` or `if (!mrktRewardToken || mrktRewardToken.length == 0)`. I think, in my case, `getOrCreateMarket` always returns a valid market, and `rewardToken = getOrCreateRewardToken()` should alwasys returns a valid rewardToken?

---

`mrkt = getOrCreateMarket(mktID)` is what I am talking about.

And I agree it will always return a Market entity. I guess my point is that if it had to make a Market entity that would be an error. It really shouldn't happen.

I usually do something like this:
```
const mrkt = Market.load(mktID);
if (!mrkt) {
     log.error();
     return;
}
```

Idk it rly isn't a major thing, I'm just being picky and we are really just informally defining standards as we do this

---

OK, makes sense. I am switching to load & check.